esphome/components/pmsx003/pmsx003.cpp

```diff
@@ -248,6 +266,13 @@ void PMSX003Component::parse_data_() {
   if (this->pm_particles_25um_sensor_ != nullptr)
     this->pm_particles_25um_sensor_->publish_state(pm_particles_25um);
 
+  // Calculate and publish AQI if sensor is configured
+  if (this->aqi_sensor_ != nullptr && this->pm_2_5_sensor_ != nullptr && this->pm_10_0_sensor_ != nullptr) {
```

Comment:
```suggestion
  if (this->aqi_sensor_ != nullptr) {
```
The values are still calculated even without these sensors being configured no point gating it